src/Cli/Microsoft.TemplateEngine.Cli/Commands/CliPathInfo.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System.Runtime.InteropServices;
+using Microsoft.DotNet.Cli.Utils;
+using Microsoft.DotNet.Configurer;
+using Microsoft.TemplateEngine.Abstractions;
+
+namespace Microsoft.TemplateEngine.Cli.Commands
+{
+    internal sealed class CliPathInfo : IPathInfo
+    {
+        public CliPathInfo(
+            ITemplateEngineHost host,
+            IEnvironment environment,
+            string? settingsLocation)
+        {
+            UserProfileDir = GetUserProfileDir(environment);
+            GlobalSettingsDir = GetGlobalSettingsDir(settingsLocation);
+
+            ValidatePathArguments(host, GlobalSettingsDir);
```

Comment:
`GlobalSettingsDir` cannot be `null`, so second part of validation is not needed.
It is preferable that validation happens as first operation in constructor, to fail immediately in case of error.